src/components/Report/ReportComponent.spec.tsx

```diff
@@ -7,257 +7,209 @@ import Adapter from 'enzyme-adapter-react-16';
-        txn_desc: 'Electricity',
-        txn_date: '2018-01-01T00:00+00:00',
-        txn_src: 'hsbc',
-        acc_sortcode: '',
-        acc_balance:0,
-        acc_number: '',
-        txn_type: 'DD',
-      },
-      {
-        txn_amount_credit: 0,
-        txn_amount_debit: 50,
-        txn_desc: 'Water',
-        txn_date: '2017-12-01T00:00+00:00',
-        txn_src: 'hsbc',
-        acc_sortcode: '',
-        acc_balance:0,
-        acc_number: '',
-        txn_type: 'DD',
-      },
-      {
-        txn_amount_credit: 0,
-        txn_amount_debit: 200,
-        txn_desc: 'Uplink Fee',
-        txn_date: '2018-02-01T00:00+00:00',
-        txn_src: 'hsbc',
-        acc_sortcode: '',
-        acc_balance: 0,
-        acc_number: '',
-        categories: [],
-      },
-      {
-        txn_amount_credit: 500,
-        txn_amount_debit: 0,
-        txn_desc: 'Bribe',
-        txn_date: '2018-06-01T00:00+00:00',
-        txn_src: 'hsbc',
-        acc_sortcode: '',
-        acc_balance: 0,
-        acc_number: '',
+  it('renders for users who have not provided a transaction password', () => {
+    const store = mockStore({
+      user: {
+        transactions: [],
         categories: [],
+        txnPassword: null,
+        token: 'abc',
+        isLoggedIn: true,
       },
-    ];
+    });
 
-    const categories: Category[] = [
-      {
-        category_rules: {
-          txn_desc: {
-            "rules": [
-              ['=', 'Hello World']
-            ]
-          }
-        },
-        name: 'Hello',
-        id: 'hello',
-        className: '',
-        txn_month_modifier: 1,
-      },
-      {
-        category_rules: {
-          txn_type: {
-            rules: [
-              ['=', 'DD']
-            ]
-          }
-        },
-        name: 'Direct Debit',
-        id: 'directdebit',
-        className: '',
-      },
-    ];
+    let component = mount(store,
+      <ReportComponent
+        date={moment('2018-01-01').utc().toDate()}
+      />
+    );
 
+    expect(component.html()).toMatchSnapshot()
+  });
+
+  it('renders for users who have provided a transaction password', () => {
     const store = mockStore({
       user: {
-        transactions: transactions,
-        categories: categories,
+        transactions: [],
+        categories: [],
         txnPassword: 'pass',
         token: 'abc',
         isLoggedIn: true,
       },
     });
 
-    enzyme.mount(
-      <Provider store={store}>
-        <BrowserRouter>
-          <ReportComponent
+    let component = mount(store,
+      <ReportComponent
+        date={moment('2018-01-01').utc().toDate()}
+      />);
+
+    expect(component.html()).toMatchSnapshot();
+  });
+
+  it('component renders for transactions', () => {
+    return new Promise((resolve, reject) => {
+      const transactions = [
+        {
+          txn_amount_credit: 0,
+          txn_amount_debit: 100,
+          txn_desc: 'Hello World',
+          txn_date: '2017-12-31T00:00:00',
```

Comment:
This had a fix for weirdness by adding `+00:00` at the end of the dates which you accidentally lost